integration_tests/tests/common/mod.rs

```diff
@@ -88,11 +105,33 @@ pub fn translate_url_for_external_access(url: &Url, external_port: u16) -> Url {
     translated
 }
 
+/// Run `test`, capturing logs from `helper` and `leader` if if fails.
+pub async fn run_test_capturing_logs<
+    Helper: CopyLogs,
+    Leader: CopyLogs,
+    Fut: Future<Output = Result<()>>,
+    Test: FnMut() -> Fut,
+>(
+    test_name: &str,
+    helper: &Helper,
+    leader: &Leader,
+    mut test: Test,
+) {
+    let result = test().await;
```

Comment:
We should probably use `std::panic::catch_unwind()` here, both to ensure we emit logs on any panics we miss, and to make writing tests in this context more ergonomic. (We can use `std::panic::panic_any()` to propagate resulting outer errors)

Reply:
I tried doing that, but `std::panic::catch_unwind` can't be passed an async closure. The async equivalent is [`futures::future::FutureExt::catch_unwind`](https://docs.rs/futures/latest/futures/future/trait.FutureExt.html#method.catch_unwind), but it requires `UnwindSafe`, which the closure we want to test is decidedly not, and so I backed off from this approach and instead made `submit_measurements_and_verify_aggregate` fallible.

Reply:
Ah right, of course. There are future wrappers that apply this to poll, FWIW: https://docs.rs/futures/latest/futures/future/struct.CatchUnwind.html